src/porepy/numerics/interface_laws/elliptic_interface_laws.py

```diff
@@ -723,4 +725,260 @@ def assemble_matrix_rhs(
+            g_l: Grid of the slave domain.
+            data_h: Data dictionary for the master domain.
+            data_l: Data dictionary for the slave domain.
+            data_edge: Data dictionary for the edge between the domains.
+
+        """
+        matrix_dictionary_edge = data_edge[pp.DISCRETIZATION_MATRICES][self.keyword]
+        parameter_dictionary_edge = data_edge[pp.PARAMETERS][self.keyword]
+
+        robin_weight = sps.block_diag(parameter_dictionary_edge["robin_weight"])
+        matrix_dictionary_edge["robin_weight"] = robin_weight
+
+    def assemble_matrix_rhs(
+        self, g_master, g_slave, data_master, data_slave, data_edge, matrix
+    ):
+        """ Assemble the dicretization of the interface law, and its impact on
+        the neighboring domains.
+        Parameters:
+        ----------
+            g_master: Grid on one neighboring subdomain.
+            g_slave: Grid on the other neighboring subdomain.
+            data_master: Data dictionary for the master suddomain
+            data_slave: Data dictionary for the slave subdomain.
+            data_edge: Data dictionary for the edge between the subdomains
+            matrix_master: original discretization for the master subdomain
+            matrix_slave: original discretization for the slave subdomain
+
+        """
+        matrix_dictionary_edge = data_edge[pp.DISCRETIZATION_MATRICES][self.keyword]
+
+        self.discretize(g_master, g_slave, data_master, data_slave, data_edge)
+
+        if not g_master.dim == g_slave.dim:
+            raise AssertionError("Slave and master must have same dimension")
+
+        master_ind = 0
+        slave_ind = 1
+
+        # Generate matrix for the coupling. This can probably be generalized
+        # once we have decided on a format for the general variables
+        mg = data_edge["mortar_grid"]
+
+        # We know the number of dofs from the master and slave side from their
+        # discretizations
+        dof = np.array(
+            [
+                matrix[master_ind, master_ind].shape[1],
+                matrix[slave_ind, slave_ind].shape[1],
+                self.ndof(mg),
+            ]
+        )
+        cc = np.array([sps.coo_matrix((i, j)) for i in dof for j in dof])
+        cc_master = cc.reshape((3, 3))
+        cc_slave = cc_master.copy()
+
+        # The rhs is just zeros
+        # EK: For some reason, the following lines were necessary to apease python
+        rhs = np.empty(3, dtype=np.object)
+        rhs[master_ind] = np.zeros(matrix[master_ind, master_ind].shape[1])
+        rhs[slave_ind] = np.zeros(matrix[slave_ind, slave_ind].shape[1])
+        rhs[2] = np.zeros(self.ndof(mg))
+
+        # The convention, for now, is to put the master grid information
+        # in the first column and row in matrix, slave grid in the second
+        # and mortar variables in the third
+        # If master and slave is the same grid, they should contribute to the same
```

Comment:
Sounds like the assembler should be reworked slightly. We can discuss this later. I presume this may also cause problems for periodic bcs for a flow problem.